platforms/software/maven/src/main/java/org/gradle/api/publish/maven/MavenPomDeploymentRepository.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.publish.maven;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.provider.Property;
+import org.jspecify.annotations.NullMarked;
+
+/**
+ * The repository information of the Maven {@code distributionManagement} node.
+ *
+ * @see MavenPom
+ * @see MavenPomDistributionManagement
+ * @since 9.1.0
+ */
+@Incubating
+@NullMarked
```

Comment:
@jvandort The package does not have this annotation so I added it here to pass the archTest. Or should I add it to the package?

Reply:
It probably _should_ be on the package. But, there are concerns with source compatibility with Kotlin, where changing something from nullable to non-nullable is not source compatible. 

We generally only change nullability markings on public types in major releases. 

For this PR, you can leave this as-is

Reply:
And what about putting the annotation to the package in 9.0.0? Is it already too late?

Reply:
Yeah no worries no reason to rush this. This will likely get addressed by an internal initiative eventually ensuring all public packages are annotated.